hydra-node/src/Hydra/Node.hs

```diff
@@ -230,7 +294,19 @@ processNextInput HydraNode{nodeState, ledger, env} e =
  where
   NodeState{modifyHeadState} = nodeState
 
-  computeOutcome = Logic.update env ledger
+  computeOutcome = HeadLogic.update env ledger
+
+processStateChanges :: MonadSTM m => HydraNode tx m -> [StateChanged tx] -> m ()
+processStateChanges node stateChanges = do
+  events <- atomically . forM stateChanges $ \stateChanged -> do
+    eventId <- getNextEventId
+    pure StateEvent{eventId, stateChanged}
+  putEventsToSinks eventSinks events
```

Comment:
This has been defined in the [ADR](https://hydra.family/head-protocol/unstable/adr/29/)

> The stepHydraNode main loop does call putEvent on all eventSinks in sequence. Any failure will make the hydra-node process terminate and require a restart.

and does not present a difference to like it was before. If the persistence failed to write to disk, it would bring down the whole node.

> do we really want to have the node crashing if a sink is not available?
What's the alternative? We can't continue if a sink failed to process an event.